config_lib/parsers/parser_yaml.py

```diff
@@ -1,11 +1,207 @@
+import re
+from datetime import datetime
+
+
 class YAMLSyntaxError(Exception):
+
     def __init__(self, message, line_num):
         super().__init__(f"YAML Syntax Error on line {line_num}: {message}")
         self.line_number = line_num
 
 
 def parse_yaml_string(yaml_str):
-    # TODO: реалізувати парсер для YAML-файла з коректною обробкою помилок.
+    lines = yaml_str.splitlines()
+    return _parse_yaml_lines(lines)
 
-    return result
 
+def _strip_inline_comment(text):
+    result = []
+    in_single_quote = False
+    in_double_quote = False
+
+    for char in text:
+        if char == "'" and not in_double_quote:
+            in_single_quote = not in_single_quote
+        elif char == '"' and not in_single_quote:
+            in_double_quote = not in_double_quote
+        elif char == '#' and not in_single_quote and not in_double_quote:
+            break
+
+        result.append(char)
+
+    return ''.join(result).strip()
+
+
+def _parse_yaml_scalar(value):
+    value = _strip_inline_comment(value)
+
+    value = value.strip()
+    if value in ('null', 'Null', 'NULL', '~'):
+        return None
+    elif value in ('true', 'True'):
+        return True
+    elif value in ('false', 'False'):
+        return False
+    elif re.match(r'^-?\d+$', value):
+        return int(value)
+    elif re.match(r'^-?\d+\.\d+$', value):
+        return float(value)
+    elif re.match(r'^\d{4}-\d{2}-\d{2}(?:[ T]\d{2}:\d{2}:\d{2})?$', value):
+        try:
+            return datetime.fromisoformat(value)
+        except ValueError:
+            return value
+    else:
+        return value.strip('"').strip("'")
```

Comment:
Remove unnecessary elif statements after a return, as they are redundant. Each conditional block can simply use if. It’s also worth adding processing for numbers with exponents. Additionally, instead of silently ignoring the ValueError, log the error or raise a more informative exception to provide better insight into issues with date parsing. It’s better to do it like this: 
```suggestion
def _parse_yaml_scalar(value):
    value = _strip_inline_comment(value).strip()

    if value in ('null', 'Null', 'NULL', '~'):
        return None
    if value in ('true', 'True'):
        return True
    if value in ('false', 'False'):
        return False
    if re.match(r'^-?\d+$', value):
        return int(value)
    if re.match(r'^-?\d+(?:\.\d+)?(?:[eE][-+]?\d+)?$', value):
        return float(value)
    if re.match(r'^\d{4}-\d{2}-\d{2}(?:[ T]\d{2}:\d{2}:\d{2})?$', value):
        try:
            return datetime.fromisoformat(value)
        except ValueError as exc:
            raise ValueError(f"Invalid date format: {value}") from exc

    return value.strip('"').strip("'")
```